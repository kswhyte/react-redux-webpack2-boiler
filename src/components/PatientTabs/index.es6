import Component from './PatientTabs';
export default Component;
