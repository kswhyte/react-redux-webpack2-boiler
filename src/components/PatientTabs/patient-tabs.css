@import "imports/_helpers.css";

.tabs-line {
  margin-top: -2px;
  display: block;
  text-align: center;
  width: 100%;
  height: 1.5px;
  background-color: var(--primary-color);
}
