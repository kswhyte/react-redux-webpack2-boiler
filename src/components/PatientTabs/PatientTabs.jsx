import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Route, Link } from 'react-router-dom';

import PatientInfo from '../PatientInfo';
import PatientAppointments from '../PatientAppointments';
import PatientNotes from '../PatientNotes';

import './patient-tabs.css';

const propTypes = {
    children: PropTypes.object,
    patientInfo: PropTypes.object,
    submitPatientInfo:PropTypes.func
  };

let tabs = {
  general : 'GENERAL',
  appointments : 'APPOINTMENTS',
  notes :'NOTES'
}

export class PatientTabs extends Component {
  constructor(props) {
    super(props);
    this.props = props;

    this.state = {
      currentTab : tabs.general
    }

    this.handleTabState = this.handleTabState.bind(this);
  }

  handleTabState(e) {
    let className = e.currentTarget.className;
    if(className.match(/general/)){
      this.setState({currentTab: tabs.general})
    } else if(className.match(/appointments/)){
      this.setState({currentTab: tabs.appointments})
    } else if(className.match(/notes/)) {
      this.setState({currentTab: tabs.notes})
    }
  }

  render(){
    return (
      <div className="patient-tabs-wrapper">
        <div className="row sm-spacer">
          <ul className="nav nav-tabs">
            <li onClick={this.handleTabState} className={`general-info-tab ${this.state.currentTab === tabs.general ? 'active' : ''}`}>
              <Link to="/patientinfo/generalinfo">General Info</Link>
            </li>
            <li onClick={this.handleTabState} className={`appointments-tab ${this.state.currentTab === tabs.appointments ? 'active' : ''}`}>
              <Link to="/patientinfo/appointments">PatientName Appointments</Link>
            </li>
            <li onClick={this.handleTabState} className={`notes-tab ${this.state.currentTab === tabs.notes ? 'active' : ''}`}>
              <Link to="/patientinfo/notes">Notes</Link>
            </li>
          </ul>

          <div className="tabs-line" />
          <div className="break-line" />
         
          <Route
            exact
            path="/patientinfo/generalinfo"
            render={() => <PatientInfo generalInfo={this.props.patientInfo.generalInfo} submitPatientInfo={this.props.submitPatientInfo} />}
          />
          <Route
            exact
            path="/patientinfo/appointments"
            render={() => <PatientAppointments appointments={this.props.patientInfo.appointments} />}
          />
          <Route exact path="/patientinfo/notes" render={() => <PatientNotes notes={this.props.patientInfo.notes} />} />
        </div>
      </div>
    )
  }
}

PatientTabs.propTypes = propTypes;

export default PatientTabs;