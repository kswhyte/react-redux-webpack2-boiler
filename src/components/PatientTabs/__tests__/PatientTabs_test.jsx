import React from 'react';
// import { storiesOf, action, linkTo } from '@kadira/storybook';
import PatientTabs from '../PatientTabs';

// import { specs, describe, it } from 'storybook-addon-specifications';

import { storiesOf, describe, it, specs } from '../../../../.storybook/facade';

import { shallow } from 'enzyme';
import expect from 'expect';

const stories = storiesOf('PatientTabs', module);

stories.add('Story', () => {
  const patientTabsStory = <PatientTabs />;

  specs(() =>
    describe('Show a successful alert', () => {
      it('Should render the PatientTabs component without crashing', () => {
        let output = shallow(patientTabsStory);
        expect(output.find('.patient-tabs-wrapper').length).toEqual(1);
      });
    })
  );
  return patientTabsStory;
});
