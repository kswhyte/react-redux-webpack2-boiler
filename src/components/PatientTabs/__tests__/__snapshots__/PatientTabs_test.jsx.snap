// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story 1`] = `
<div
  className="patient-tabs-wrapper"
>
  <div
    className="row sm-spacer"
  >
    <ul
      className="nav nav-tabs"
    >
      <li
        className="general-info-tab active"
        onClick={[Function]}
      >
        <a
          href="/patientinfo/generalinfo"
          onClick={[Function]}
        >
          General Info
        </a>
      </li>
      <li
        className="appointments-tab "
        onClick={[Function]}
      >
        <a
          href="/patientinfo/appointments"
          onClick={[Function]}
        >
          PatientName Appointments
        </a>
      </li>
      <li
        className="notes-tab "
        onClick={[Function]}
      >
        <a
          href="/patientinfo/notes"
          onClick={[Function]}
        >
          Notes
        </a>
      </li>
    </ul>
    <div
      className="tabs-line"
    />
    <div
      className="break-line"
    />
  </div>
  <div
    className="row sm-spacer"
  >
    <div
      className="break-line"
    />
  </div>
  <div
    className="row xs-spacer sm-spacer"
  >
    <div
      className="col-sm-6 no-padding"
    >
      <button
        className="back-btn primary-btn long"
        tabIndex={1}
      >
        BACK
      </button>
    </div>
    <div
      className="col-sm-6 no-padding"
    >
      <button
        className="next-btn primary-btn long"
        tabIndex={1}
        type="submit"
      >
        NEXT
      </button>
    </div>
  </div>
</div>
`;
